core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -387,7 +387,15 @@ class SparkContext(config: SparkConf) extends Logging {
     require(SparkContext.VALID_LOG_LEVELS.contains(upperCased),
       s"Supplied level $logLevel did not match one of:" +
         s" ${SparkContext.VALID_LOG_LEVELS.mkString(",")}")
-    Utils.setLogLevel(Level.toLevel(upperCased))
+    // Update only if new log level is not same as current log level
+    if (upperCased != Utils.getLogLevel) {
```

Comment:
What if "spark.log.level" was not use but the executor log level is different from the driver based on the log4 configs? One might try to call `sc.setLogLevel` but because of this check the new log level might not reach the executor if he calls it with the current driver log level.

So I would move the following part outside of this outer if:
```
      if (conf.get(EXECUTOR_ALLOW_SYNC_LOG_LEVEL) && _schedulerBackend != null) {
        _conf.set(SPARK_LOG_LEVEL.key, upperCased)
        _schedulerBackend.update(upperCased)
      }
```